### PR TITLE
Added and tested yolov8s and yolov8n model

### PR DESCRIPTION
**Pull request by cabelo**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [X] There is a reference to the original bug report and related work
- [X] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [X] The feature is well documented and sample code can be built with the project CMake

Below is evidence of the test:
yolos-n

### CODE REVIEW

File: samples/dnn/models.yml [hunk 1]
```diff
@@ -34,6 +34,36 @@ yolov8x:
...
+  width: 640
+  height: 640
+  rgb: true
+  classes: "object_detection_classes_yolo.txt"
+  background_label_id: 0
+  sample: "yolo_detector"
+
+yolov8n:
+  load_info:
+    url: "https://huggingface.co/cabelo/yolov8/resolve/main/yolov8n.onnx?download=true"
```
Comment (dkurt) [inline samples/dnn/models.yml:53]: We usually don't accept personal fileshare

**Comment by cabelo** [comment]
Hi @dkurt,

Thanks for the return message. I made this PR to make it easier for lay users. I receive many emails requesting this.

I made this available because I didn't find ready-made yolov8 models in onnx available for download on the internet anywhere.

**Comment by cabelo** [comment]
Hi @dkurt,

A humble suggestion: Is there any procedure for me to upload the models and obtain a reliable link for you?

**Comment by dkurt** [comment]
@cabelo, can you please that the same sample works with https://github.com/CVHub520/X-AnyLabeling/releases/download/v0.1.0/yolov8s.onnx and https://github.com/CVHub520/X-AnyLabeling/releases/download/v0.1.0/yolov8n.onnx ? If so, I recommend to use them instead.

**Comment by cabelo** [comment]
> @cabelo, can you please check that the same sample works with https://github.com/CVHub520/X-AnyLabeling/releases/download/v0.1.0/yolov8s.onnx and https://github.com/CVHub520/X-AnyLabeling/releases/download/v0.1.0/yolov8n.onnx ? If so, I recommend to use them instead.

I'll do it tonight, thanks for the tip.

**Comment by cabelo** [comment]
Hi @dkurt,
Thanks for the tip and patience, the 2 links were tested ( https://github.com/CVHub520/X-AnyLabeling/releases/download/v0.1.0/yolov8s.onnx and https://github.com/CVHub520/X-AnyLabeling/releases/download/v0.1.0/yolov8n.onnx ), below is the evidence image.


opencv

**Review by dkurt — APPROVED** [review]

**Pull request merged**